README.md

```diff
@@ -1,6 +1,56 @@
 # CWMSpy
 CWMS REST API for Data Retrieval
 
+## Development
+
+In order to set up the development environment you will need to have [poetry][poetry] installed on your computed. To install all of the project's dependencies, run the following command:
+
+```sh
+poetry install
+```
+
+This will create a virtual environment in `.venv/` in the project's root directory.
+
+
+### Running Tests and Type Checker
+
+The following commands can be used to run the test suite and the static type checker.
+
+```sh
+# Run all tests
+poetry run pytest -v tests/
+
+# Check types
+poetry run mypy --strict cwms/
+```
+
+Contributors are encouraged to run the tests and type checker before making commits, since both of these are included in the CI pipeline.
+
+
+### Contributing
+
+In order for code to be committed to the repo it must be formatted using [black][black] and [isort][isort]. Developers are encouraged to integrate these tools into their workflow. Run the following command to install both tools as pre-commit hooks. Once installed, all staged source files will be automatically formatted before being committed.
```

Comment:
Is there a validation mechanism that could be put in place for a GitHub Check to get added to the PR?

Reply:
I have a GitHub Actions workflow that runs `black` and `isort` on all `push` and `pull_request` events, if that's what you are referring to. I was going to wait to add it until the codebase had been reformatted, but I can add it in this pull request if you think it should be included.

Reply:
I'm not generally a fan of the CI/CD editing code on my behalf and would prefer a Check to validate that I had run the linters locally.

Reply:
Sorry for the confusion, the workflow is a check only. Only the pre-commit hooks will actually modify the files (if they are installed).

Reply:
ah gotcha, adding in a separate PR makes sense. Don't want to format everything in this initial PR